### PR TITLE
Feature/56718 routing

### PR DESCRIPTION
**Pull request by kleyla**
## Description

Routing added.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
| ✓ | :hammer: Refactoring       |
|  ✓ | :100: Add tests            |
|   | :link: Update dependencies |
|   | :scroll: Docs              |

### CODE REVIEW

File: src/components/error/Error404.class.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export class Error404 {
+  text: string;
+
+  constructor() {
+    this.text = 'Error 404';
```
Comment (marcelo-yourday) [inline src/components/error/Error404.class.ts:5]: Better if we use a friendly message here, something like: "Página no encontrada".

File: src/router/router.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+      if (routePathSegs.length !== pathSegments.length) {
+        return false;
+      }
+      return routePathSegs.every((routePathSeg, i) => {
+        return routePathSeg == pathSegments[i];
+      });
+    });
+    if (!matchedRoute) {
+      return this.routes[1];
```
Comment (marcelo-yourday) [inline src/router/router.ts:41]: Avoid using magic numbers. Maybe it is better to get the route index from the routes.ts file.

File: src/router/router.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  matchUrlToRoute(pathSegments: string[]): RouteType {
+    const matchedRoute = this.routes.find((route) => {
+      const routePathSegs = route.path.split('/').slice(1);
+
+      if (routePathSegs.length !== pathSegments.length) {
+        return false;
+      }
+      return routePathSegs.every((routePathSeg, i) => {
+        return routePathSeg == pathSegments[i];
```
Comment (marcelo-yourday) [inline src/router/router.ts:37]: We should try to use the === instead the ==

**Review by marcelo-yourday — CHANGES_REQUESTED** [review]

File: src/router/router.ts (hunk 3 above)
Comment (kleyla) [reply src/router/router.ts]: Thank you, changes are done and already reached the 80% of code coverage.

**Review by marcelo-yourday — APPROVED** [review]

**Pull request merged**